fs2/src/main/scala/laserdisc/fs2/RedisClient.scala

```diff
@@ -8,57 +8,76 @@ import cats.syntax.all._
+    * redis node and handles the blocking network
+    * connection's operations on a cached thread pool.
+    */
+  @inline final def to[F[_]: LogWriter: ContextShift: Timer: Concurrent](
+      host: Host,
+      port: Port,
+      writeTimeout: Option[FiniteDuration] = Some(10.seconds),
+      readMaxBytes: Int = 256 * 1024
+  ): Resource[F, RedisClient[F]] =
+    Blocker[F] >>= (blockingPool => blockingOn(blockingPool)(Set(RedisAddress(host, port)), writeTimeout, readMaxBytes))
+
   /**
     * Creates a redis client that will handle the blocking network
     * connection's operations on a cached thread pool.
     */
-  @inline final def apply[F[_]: LogWriter: Concurrent: ContextShift: Timer](
+  @inline final def toNodes[F[_]: LogWriter: ContextShift: Timer: Concurrent](
       addresses: Set[RedisAddress],
       writeTimeout: Option[FiniteDuration] = Some(10.seconds),
       readMaxBytes: Int = 256 * 1024
   ): Resource[F, RedisClient[F]] =
     Blocker[F] >>= (blockingPool => blockingOn(blockingPool)(addresses, writeTimeout, readMaxBytes))
 
   /**
-    * Creates a redis client for a single redis node
-    * that will handle the blocking network connection's
-    * operations on a cached thread pool.
+    * Creates a redis client that connects to a single
+    * redis node and handles the blocking network
+    * connection's operations on a cached thread pool.
     */
-  @inline final def toNode[F[_]: LogWriter: Concurrent: ContextShift: Timer](
+  @inline final def toNodeBlockingOn[F[_]: LogWriter: ContextShift: Timer: Concurrent](
+      blockingPool: Blocker
+  )(
       host: Host,
       port: Port,
       writeTimeout: Option[FiniteDuration] = Some(10.seconds),
       readMaxBytes: Int = 256 * 1024
   ): Resource[F, RedisClient[F]] =
-    Blocker[F] >>= (blockingPool => blockingOn(blockingPool)(Set(RedisAddress(host, port)), writeTimeout, readMaxBytes))
+    blockingOn(blockingPool)(Set(RedisAddress(host, port)), writeTimeout, readMaxBytes)
 
   /**
-    * Creates a redis client for a single redis node
-    * that will handle the blocking network connection's
-    * operations on a cached thread pool.
+    * Creates a redis client that connects to a single
+    * redis node and handles the blocking network
+    * connection's operations on a cached thread pool.
+    * The client will generate no logs (uses a noOpLog)
     */
-  @inline final def toNodeBlockingOn[F[_]: LogWriter: Concurrent: ContextShift: Timer](
-      blockingPool: Blocker
-  )(
+  @inline final def toNodeNoLogs[F[_]](
       host: Host,
       port: Port,
       writeTimeout: Option[FiniteDuration] = Some(10.seconds),
       readMaxBytes: Int = 256 * 1024
+  )(
+      implicit cs: ContextShift[F],
+      ti: Timer[F],
+      co: Concurrent[F]
   ): Resource[F, RedisClient[F]] =
-    blockingOn(blockingPool)(Set(RedisAddress(host, port)), writeTimeout, readMaxBytes)
+    to(host, port, writeTimeout, readMaxBytes)(SyncLogWriter.noOpLog[F], cs, ti, co)
```

Comment:
It feels wrong to have a constructor that exists solely to select a logging implementation.  If you need to add other constructors for actual redis-related stuff, now you'll probably have to add additional `noLog` constructors too?  

I'm probably going to get shot for this, but would it work to just leave toNode alone, but provide noOpLog as a default value for the LogWriter impl?

Reply:
I went with this because I didn't want to pass the message that the default (or suggested) use of the client is without logs (that's mainly needed for tests actually). Anyway if it's mostly used without logs even in production then it makes even more sense to have it as a default. I'm easy on this, happy to change it.

Reply:
Well, it'd be nice if it just picked up the presence of a logging framework but I guess that doesn't fit so hot with compile time safety :)

Reply:
It makes sense actually. That would be good. I think I might have a version that tries to use a `LogWriter` if it's in scope and defaults to a silent log if it isn't. I'll push it here.